planet/sync/client.py

```diff
@@ -37,17 +37,27 @@ class Planet:
     Parameters:
         session: Optional Session. The Session can be provided allowing for customization, and
             will default to standard behavior when not provided.
+        data_base_url: Optional base URL for Data API. Defaults to production.
+        orders_base_url: Optional base URL for Orders API. Defaults to production.
+        subscriptions_base_url: Optional base URL for Subscriptions API. Defaults to production.
+        features_base_url: Optional base URL for Features API. Defaults to production.
 
     """
 
-    def __init__(self, session: Optional[Session] = None) -> None:
+    def __init__(self,
+                 session: Optional[Session] = None,
+                 data_base_url: Optional[str] = None,
+                 orders_base_url: Optional[str] = None,
+                 subscriptions_base_url: Optional[str] = None,
+                 features_base_url: Optional[str] = None) -> None:
```

Comment:
We should be mindful that this will become part of the SDK API that users are presented with, but users may not have a need for this level of configuration. We should try to keep our API surface clean and simple if possible.

Another approach could be to accept a single, common `base_url` (defaulting to `https://api.planet.com`) and each class under the hood (DataAPI, OrdersAPI) etc can add the appropriate path suffix e.g. `/data/v1`.

Reply:
Agreed a single configuration would be ideal. It would also be nice to allow specification via CLI.

Also, since the CLI uses the async clients, we'd need to alter the approach taken there of statically building the URLS (like in data.py) based off of the `constants` module